src/gui/dialogs/unit_recall.cpp

```diff
@@ -181,6 +181,13 @@ void unit_recall::pre_show(window& window)
 		find_widget<button>(&window, "show_help", false),
 		std::bind(&unit_recall::show_help, this, std::ref(window)));
 
+	list.register_sorting_option(0, [this](const int i) { return recall_list_[i]->level(); });
```

Comment:
You're duplicating the registration of sorting options. Please don't do that: only edit the existing sorting lambdas in lines 240-246.